crates/parsing/src/grammar/rules.rs

```diff
@@ -537,6 +560,13 @@ fn expr_atom(parser: &mut Parser) {
             expr_record(parser);
             return expression.end(parser, SyntaxKind::LiteralExpression);
         }
+        SyntaxKind::Underscore => {
+            parser.consume();
+            // It's called `ExprSection` in the purs compiler, and I just stole that name because I
+            // couldn't come up with anything that felt right. `ImplicitFunctionExpression`,
+            // `UnderscoreExpression`?
+            return expression.end(parser, SyntaxKind::SectionExpression);
```

Comment:
The naming here is tricky because the underscore itself affects how its parents are interpreted. I'm fine keeping it consistent with the compiler but maybe that other name could be what we call "expressions that have sections"

Reply:
Okay, we'll leave it as SectionExpression.

I am aware that the parsing some parts of PureScript is extra tricky - it's going to be a challenge to be fully compatible with the mainline compiler.